install/resources/upgrade34.php

```diff
@@ -0,0 +1,46 @@
+<?php
+/**
+ * MyBB 1.8
+ * Copyright 2014 MyBB Group, All Rights Reserved
+ *
+ * Website: http://www.mybb.com
+ * License: http://www.mybb.com/about/license
+ *
+ */
+
+/**
+ * Upgrade Script: 1.8.5
+ */
+
+$upgrade_detail = array(
+	"revert_all_templates" => 0,
+	"revert_all_themes" => 0,
+	"revert_all_settings" => 0
+);
+
+@set_time_limit(0);
+
+function upgrade34_dbchanges()
+{
+	global $db, $output;
+
+	$query = $db->simple_select('settings', 'value', "name='username_method'");
```

Comment:
> The $settings array is still build from a cache file which content may differ from that of the DB.

https://github.com/mybb/mybb/blob/feature/install/upgrade.php#L94-128 - what exactly will change the setting value between running this code and the 1.8.6 upgrade part (ACP and front-end are locked, just to remind)? Could you show an example?

> BTW those references only make use of the setting value not actually update values as this case does.

?
The 1st example also runs a query that updates stuff in database based on the setting's value. I'm of course only talking about the redundacy of `simple_select()`, not the update queries.